pulp_ansible/app/serializers.py

```diff
@@ -361,151 +373,304 @@ class Meta:
+    )
 
-    files = serializers.JSONField(help_text=_("A JSON field holding FILES.json data."))
+    files = serializers.JSONField(
+        help_text=_("A JSON field holding FILES.json data."), read_only=True
+    )
 
     documentation = serializers.CharField(
-        help_text=_("The URL to any online docs."), allow_blank=True, max_length=2000
+        help_text=_("The URL to any online docs."),
+        allow_blank=True,
+        max_length=2000,
+        read_only=True,
     )
 
     homepage = serializers.CharField(
         help_text=_("The URL to the homepage of the collection/project."),
         allow_blank=True,
         max_length=2000,
+        read_only=True,
     )
 
     issues = serializers.CharField(
-        help_text=_("The URL to the collection issue tracker."), allow_blank=True, max_length=2000
+        help_text=_("The URL to the collection issue tracker."),
+        allow_blank=True,
+        max_length=2000,
+        read_only=True,
     )
 
     license = serializers.ListField(
         help_text=_("A list of licenses for content inside of a collection."),
         child=serializers.CharField(max_length=32),
+        read_only=True,
     )
 
-    name = serializers.CharField(help_text=_("The name of the collection."), max_length=64)
+    name = serializers.CharField(
+        help_text=_("The name of the collection."), max_length=64, read_only=True
+    )
 
     namespace = serializers.CharField(
-        help_text=_("The namespace of the collection."), max_length=64
+        help_text=_("The namespace of the collection."), max_length=64, read_only=True
     )
 
-    repository = serializers.CharField(
-        help_text=_("The URL of the originating SCM repository."), allow_blank=True, max_length=2000
+    origin_repository = serializers.CharField(
+        help_text=_("The URL of the originating SCM repository."),
+        source="repository",
+        allow_blank=True,
+        max_length=2000,
+        read_only=True,
     )
 
     tags = TagNestedSerializer(many=True, read_only=True)
 
-    version = serializers.CharField(help_text=_("The version of the collection."), max_length=128)
+    version = serializers.CharField(
+        help_text=_("The version of the collection."), max_length=128, read_only=True
+    )
 
     requires_ansible = serializers.CharField(
         help_text=_(
             "The version of Ansible required to use the collection. "
             "Multiple versions can be separated with a comma."
         ),
         allow_null=True,
-        required=False,
+        read_only=True,
         max_length=255,
     )
 
+    creating = True
+
+    def validate(self, data):
+        """Run super() validate if creating, else return data."""
+        if self.creating:
+            return super().validate(data)
+        return data
+
+    def is_valid(self, raise_exception=False):
+        """
+        Allow this serializer to be used for validating before saving a model.
```

Comment:
I'm still not sure we need this. Is it there to support code we plan to deprecate/delete? If i read correctly it allows the serializer to consume validated data, right? I do not think they are designed that way.
My thinking is if we hand the same raw request data that we get in the viewset to the task, then the serializer in the task can validate in the same way plus the additional deferred_validate step. No need for any serializer field to behave differently.
Not blocking on this now. But can you confirm that this will fall for another refactor to come?

Reply:
Yes it most likely will be removed once the `import_collection` code is finally removed. `read_only` fields are not validated when initialized in a serializer and they don't appear in the `validate` method.